psydac/pyccel/ast/core.py

```diff
@@ -0,0 +1,5703 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import importlib
+from collections.abc import Iterable
+from collections     import OrderedDict
+
+from sympy import sympify
+from sympy import Add as sp_Add, Mul as sp_Mul, Pow as sp_Pow
+from sympy import Integral, Symbol, Tuple
+from sympy import Lambda
+from sympy import Integer as sp_Integer
+from sympy import Float as sp_Float, Rational as sp_Rational
+from sympy import preorder_traversal
+
+from sympy.simplify.radsimp   import fraction
+from sympy.core.compatibility import with_metaclass
+from sympy.core.assumptions   import StdFactKB
+from sympy.core.singleton     import Singleton, S
+from sympy.core.function      import Function, Application
+from sympy.core.function      import Derivative, UndefinedFunction as sp_UndefinedFunction
+from sympy.core.function      import _coeff_isneg
+from sympy.core.expr          import Expr, AtomicExpr
+from sympy.logic.boolalg      import And as sp_And, Not as sp_Not, Or as sp_Or
+from sympy.logic.boolalg      import Boolean as sp_Boolean
+from sympy.tensor             import Idx, Indexed, IndexedBase
+
+from sympy.matrices.matrices            import MatrixBase
+from sympy.matrices.expressions.matexpr import MatrixSymbol, MatrixElement
+from sympy.tensor.array.ndim_array      import NDimArray
+from sympy.utilities.iterables          import iterable
+from sympy.utilities.misc               import filldedent
+
+
+from .basic     import Basic, PyccelAstNode
+from .builtins  import Enumerate, Len, List, Map, Range, Zip, PythonTuple, PythonBool, PythonInt
+from .datatypes import (datatype, DataType, CustomDataType, NativeSymbol,
+                        NativeInteger, NativeBool, NativeReal,
+                        NativeComplex, NativeRange, NativeTensor, NativeString,
+                        NativeGeneric, NativeTuple, default_precision)
+
+from .numbers        import BooleanTrue, BooleanFalse, Integer as Py_Integer, ImaginaryUnit
+from .functionalexpr import GeneratorComprehension as GC
+from .functionalexpr import FunctionalFor
+
+from ..errors.errors import Errors
+from ..errors.messages import *
+
+errors = Errors()
+
```

Comment:
We don't need to copy all the content of the file, we only keep what we import from this file

Reply:
Yes, this can be done for every file in principle. But I don't think it is a useful investment of time given that this is an "intermediate" temporary solution. 😉

Reply:
Also, given that we use the `import *` syntax, there is no simple way to know what we really use from that module.